mobile/lib/features/trade/trade_bottom_sheet_tab.dart

```diff
@@ -59,6 +59,39 @@ class _TradeBottomSheetTabState extends State<TradeBottomSheetTab> {
     tentenoneConfigChangeNotifier = context.read<TenTenOneConfigChangeNotifier>();
     positionChangeNotifier = context.read<PositionChangeNotifier>();
 
+    // init the short trade values
+    final shortTradeValues = provider.fromDirection(Direction.short);
+    shortTradeValues.updateQuantity(shortTradeValues.maxQuantity);
+    // overwrite any potential pre-existing state
+    shortTradeValues.openQuantity = Usd.zero();
+
+    // by default we set the amount to the max quantity.
+    shortTradeValues.updateContracts(shortTradeValues.maxQuantity);
+
+    // init the long trade values
+    final longTradeValues = provider.fromDirection(Direction.long);
+    longTradeValues.updateQuantity(longTradeValues.maxQuantity);
+    // overwrite any potential pre-existing state
+    longTradeValues.openQuantity = Usd.zero();
+
+    // by default we set the amount to the max quantity.
+    longTradeValues.updateContracts(longTradeValues.maxQuantity);
+
+    if (positionChangeNotifier.positions.containsKey(contractSymbol)) {
+      // in case there is an open position we have to set the open quantity for the trade values of
+      // the opposite direction
+      final position = positionChangeNotifier.positions[contractSymbol]!;
+      final tradeValues = provider.fromDirection(position.direction.opposite());
+
+      tradeValues.openQuantity = position.quantity;
```

Comment:
🔧
Not you who introduced this pattern: it might be better to start using `setters` on `TradeValues` it's getting a bit hard to see where things are being updated.